guides/developer/plugin-creators.md

```diff
@@ -0,0 +1,244 @@
+const plugin = {
+  initialize: init,
+};
+
+// Call spinnaker settings to actually load the stage
+// plugin for us
+window.spinnakerSettings.onPluginLoaded(plugin);
+```
+
+**Render Method**
+Anything can go in the render method. What is in the render method will be shown to plugin users when configuring their Spinnaker pipeline. In this example, the user can input the maximum number of seconds to wait to continue executing the pipeline.
+
+**Set Methods**
+Stages are made up of JSON that contains all information that will be passed to the backend. To update the stage JSON with the data the user enters use `this.props.updateStageField` method that takes in a valid JSON object of what to update. In this example we are updating the `maxWaitTime` field with the value that the user enters.
+
+**Register Stage**
+The `registerStage` method is what makes the stage available to be used. These are the required fields for registering a stage.
+
+1. key → a unique name of the stage
+2. label → is what is used inside the UI to display, saying what the name of the stage is
+3. description → a short description of what the stage will do
+4. component → if using React to create a stage, this is where you would put the component to render
+
+Optional Fields:
+
+1. cloudProvider → if the stage can only be ran in one of the cloud providers, that can be selected here
+# Writing The Plugin Manifest
+
+Here is an example of a plugin manifest:
+```
+name: armory/s3copy
+description: Copies S3 files to different locations
+manifestVersion: plugins/v1
+version: 1.2.3
+options:
+  s3:
+    username: user
+    password: pass
+resources:
+  orca:
+  - https://stage-plugin-test.s3-us-west-2.amazonaws.com/stage-plugin-0.0.1-SNAPSHOT.jar
+  deck:
+  - https://stage-plugin-test.s3-us-west-2.amazonaws.com/stage-plugin-ui-0.0.1-SNAPSHOT.js
+```
+
+The `name` is the name of the plugin that is being written. Names are namespaced so that plugins can have the same name, but be by different vendors. In this case the namespace is `armory` and the name of the plugin is `s3copy`.
+
+The `description` gives the plugin user an idea of what the plugin will be doing.
+
+Plugin manifests can change overtime, the `manifestVersion` tells Spinnaker what version to use to validate the manifest. Currently there is only the `plugins/v1` version.
+
+`version` is the version of the plugin. 
+
+Plugin users may want to change some settings to control how the plugin works. For example controlling what username and password to use to connect to S3. The `options` key gives the plugin users that flexibility. Anything under `options` the plugin user can modify.
+
+The next section in the manifest is for `resources`. Resources are things that are required for the plugin to run. For example when creating a stage there will be jar(s) and Javascript code that needs to be consumed by the plugin user. Currently there are two different types of `resources`. The first is for `orca`. 
```

Comment:
What's the second kind of resource? It's weird to say "The first is x" without saying "The second is y" after saying there are 2.